samcli/lib/remote_invoke/remote_invoke_executors.py

```diff
@@ -0,0 +1,224 @@
+"""
+Abstract class definitions and generic implementations for remote invoke
+"""
+import json
+import logging
+from abc import ABC, abstractmethod
+from io import TextIOWrapper
+from pathlib import Path
+from typing import Any, Callable, List, Optional, Union, cast
+
+LOG = logging.getLogger(__name__)
+
+
+class RemoteInvokeExecutionInfo:
+    """
+    Keeps request and response information about remote invoke execution
+
+    payload: payload string given by the customer
+    payload_file: if file is given, this points to its location
+
+    response: response object returned from boto3 action
+    exception: if an exception is thrown, it will be stored here
+    """
+
+    # Request related properties
+    payload: Optional[Union[str, List, dict]]
+    payload_file: Optional[TextIOWrapper]
+
+    # Response related properties
+    response: Optional[Union[dict, str]]
+    exception: Optional[Exception]
+
+    def __init__(self, payload: Optional[Union[str, List, dict]], payload_file: Optional[TextIOWrapper]):
+        self.payload = payload
+        self.payload_file = payload_file
+        self.response = None
+        self.exception = None
+
+    def is_file_provided(self):
+        return self.payload_file
```

Comment:
what is the return type of this method? .. if it is boolean, could you please change the return statement to be `bool(elf.payload_file)`

Reply:
Also add the return type to the function definition